### PR TITLE
[Exchange Oracle] Add `iou_threshold` for polygon tasks and enable single shape mode

### PR DESCRIPTION
**Pull request by Bobronium**
## Context behind the change
Allow `iou_threshhold` setting to be configured for polygons. Default value is 0.5 which should be sufficient for the most tasks. Increasing the value will require more precision and will likely cause more false-negatives. It might make sense to increase it for segmentation tasks, though experimentation is needed.

## How has this been tested?
Test process was similar to one described in #2722.

## Release plan; Potential risks; What to monitor; Rollback plan
Doesn't introduce and breaking changes, DB migrations nor public API changes.

### CODE REVIEW

File: packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py [hunk 1]
```diff
@@ -550,6 +550,12 @@ def build(self):
         return super().build()
 
 
+class PolygonTaskBuilder(SimpleTaskBuilder):
+    def _setup_quality_settings(self, task_id: int, **overrides) -> None:
+        values = {"iou_threshold": Config.core_config.polygons_iou_threshold, **overrides}
```
Comment (zhiltsov-max) [inline packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py:555]:
```suggestion
        values = {"iou_threshold": Config.core_config.polygons_iou_threshold}
        values.update(**overrides)
```
The code can fail if there is `iou_threshold` in `overrides`.

File: packages/examples/cvat/exchange-oracle/src/core/config.py [hunk 2]
```diff
@@ -258,6 +258,9 @@ class CoreConfig:
     min_roi_size_h = int(os.environ.get("MIN_ROI_SIZE_H", 300))
     "Minimum absolute ROI size for image_boxes_from_points and image_skeletons_from_boxes tasks"
 
+    polygons_iou_threshold = float(os.environ.get("POLYGONS_IOU_THRESHOLD", 0.8))
```
Comment (zhiltsov-max) [inline packages/examples/cvat/exchange-oracle/src/core/config.py:261]: Please move it into the `CvatConfig` class, next to other similar variables.

File: packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py (hunk 1 above)
Comment (Bobronium) [reply packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py:555]: Python allows duplicate keys in dict literal, the last key will remain so result would be identical in this case.

```python
print({"foo": 1, "foo": 2})  # {"foo": 2}
print({"foo": 1, **{"foo": 2}})  # {"foo": 2}
```

You might have confused this with function arguments expansion in function calls (usually `**kwargs`), in this case, python indeed doesn't allow multiple arguments to be present.

File: packages/examples/cvat/exchange-oracle/src/core/config.py (hunk 2 above)
Comment (Bobronium) [reply packages/examples/cvat/exchange-oracle/src/core/config.py]: Good catch. Done.

File: packages/examples/cvat/exchange-oracle/src/core/config.py [hunk 3]
```diff
@@ -195,6 +195,9 @@ class CvatConfig:
     cvat_iou_threshold = float(os.environ.get("CVAT_IOU_THRESHOLD", 0.8))
     cvat_oks_sigma = float(os.environ.get("CVAT_OKS_SIGMA", 0.1))
 
+    cvat_polygons_iou_threshold = float(os.environ.get("CVAT_POLYGONS_IOU_THRESHOLD", 0.5))
```
Comment (zhiltsov-max) [inline packages/examples/cvat/exchange-oracle/src/core/config.py:198]: Please update `.env.template` with the new variable

Comment (Bobronium) [reply packages/examples/cvat/exchange-oracle/src/core/config.py:198]: Done

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**